### ISSUE TITLE
Invalid dest causes runaway node process

### ISSUE THREAD

**Issue by roncli** [expressjs/multer#139]
Given the following code:

```
var express = require("express"),
    multer = require("multer"),
    app = express();

app.use(multer({dest: "h:\\drive\\does\\not\\exist"}));
```

I would expect an exception to be thrown stating that the destination does not exist.

Instead, the node process goes to 100% CPU and starts a runaway memory leak that very quickly locks up the system if not killed.

**Comment by LinusU** [comment]
Hmm, I guess that this is an windows only issue. I tried it on OS X and it created the directories necessary. Since UNIX dosen't have the concept of drives there isn't a way to replicate this.

I'll attach my test script below if it helps someone.

```javascript
var express = require('express')
var multer = require('multer')

var app = express()
var upload = multer({ dest: '/tmp/why/u/no/exist' })

app.get('/', function (req, res) {
  res.setHeader('Content-Type', 'text/html')
  res.write('<form action="/" method="POST" enctype="multipart/form-data">')
  res.write('  <input type="file" />')
  res.write('  <input type="submit" />')
  res.write('</form>')
  res.end()
})

app.post('/', upload, function (req, res) {
  res.end('It worked!')
})

app.listen(3000)
console.log('http://localhost:3000')
```

-----

@roncli Could you try the following script on Windows and tell me what it outputs?

```javascript
var mkdirp = require('mkdirp')

mkdirp('h:\\why\\u\\no\\exist', function (err) {
  if (err) throw err;

  console.log('It worked!')
})
```

It will need `mkdirp` (`npm install mkdirp`), thanks!

**Comment by roncli** [comment]
Note, I have an H drive.  I don't have a Q drive.  H drive succeeded, Q drive started sucking down memory.  Looks like the issue is with mkdirp, and conveniently they have an issue open about this already, substack/node-mkdirp#70 - Thanks for your time!

```
> var mkdirp = require("mkdirp");
undefined
> mkdirp("h:\\why\\u\\no\\exist", function(err) { if (err) throw err; console.log("It worked!"); });
undefined
> It worked!
> mkdirp("q:\\why\\u\\no\\exist", function(err) { if (err) throw err; console.log("It worked!"); });
undefined
>
```

**Comment by LinusU** [comment]
No problem, let me know if they fix it and I'll bump the dependency :+1:

**Comment by roncli** [comment]
I'm going to reopen this, because the dependency node-mkdirp is no longer supported.  Check out https://github.com/sindresorhus/make-dir for a replacement. :)

**Comment by jonchurch** [comment]
@roncli I pushed a PR with your suggested change, if you feel like testing it out I'd be much obliged 🙏

**Comment by jonchurch** [comment]
Opened a PR w/ mkdirp to fix this on their 0.x branch https://github.com/isaacs/node-mkdirp/pull/15

**Comment by jonchurch** [comment]
This has been resolved with the publishing of `mkdirp 0.5.4` 🎉
